### ISSUE TITLE
no-debugging-utils is too sensitive

### ISSUE THREAD

**Issue by TrueWill** [testing-library/eslint-plugin-testing-library#544]
### What rule do you want to change?

no-debugging-utils

### Does this change cause the rule to produce more or fewer warnings?

Fewer warnings

### How will the change be implemented?

Better RegEx?

### Example code

Currently `log.debug('message')` in production code is flagged by default. This is using the loglevel library, but many logging frameworks have a debug level, and it's common to leave these in code (as the library will filter them out by default).

### How does the current rule affect the code?

It flags the above statement.

### How will the new rule affect the code?

It will not flag the above statement.

### Do you want to submit a pull request to change the rule?

No

**Comment by Belco90** [comment]
Hey @TrueWill! This is caused by the Aggressive Reporting feature. Have you checked [the Troubleshooting section](https://github.com/testing-library/eslint-plugin-testing-library#troubleshooting) to find more about restricting this behavior?

In your case, I'd suggest two options:

1. If you are getting false positives outside testing files, [override ESLint main config so you run `eslint-plugin-testing-library` only against your testing files](https://github.com/testing-library/eslint-plugin-testing-library#eslint-overrides).
2. If you are getting false positives within testing files too, [you can indicate to `eslint-plugin-testing-library` which modules you want to consider valid to get the errors reported from](https://github.com/testing-library/eslint-plugin-testing-library#testing-libraryutils-module).

I hope one of these options can help you solve your issues.

**Comment by TrueWill** [comment]
@Belco90 Thank you; I had not seen that section.

I was not able to get this to work. If I add an `overrides` section to my .eslintrc.json it still applies the rule to files that do not match the glob. I've double-checked this, revised the glob, etc. without success.

I'll close this issue, though, as Troubleshooting discusses it and that rule isn't one our team needs.

**Comment by Belco90** [comment]
@TrueWill sad to hear it didn't fix it. I would be able to take a better look if you could provide some extract of your configs, or a small repo.

**Comment by TrueWill** [comment]
@Belco90 I think I figured it out.

Here's my (simplified) .eslintrc.json with the fix:

```json
{
  "env": {
    "browser": true,
    "es2021": true,
    "jest": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:testing-library/react",
    "prettier"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "plugins": ["@typescript-eslint", "testing-library"],
  "overrides": [
    {
      "files": ["**/specs/**/*.[jt]s?(x)"],
      "extends": ["plugin:testing-library/react"],
      "rules": {
        "testing-library/no-node-access": "off",
        "testing-library/no-container": "off"
      }
    }
  ],
  "rules": {
    "@typescript-eslint/no-empty-function": "off",
    "@typescript-eslint/no-node-access": "off",
    "@typescript-eslint/no-non-null-assertion": "off",
    "testing-library/no-debugging-utils": "off"
  }
}
```

This only runs the no-debugging-utils part against the JS/TS files under /specs.

What I didn't realize before was that the _rules still apply_ to other files. I believe you have to explicitly disable them in the top-level `rules` section (in this case with `"testing-library/no-debugging-utils": "off"`).

I think you also have to specify an `overrides`-level `rules` section with the rules for those specific files (the tests).

This wasn't obvious to me from the Troubleshooting section or from the ESLint documentation.

**Comment by Belco90** [comment]
I see. You are linting all other files indeed. Rather than switching `"testing-library/no-debugging-utils"` off in the `rules` property, you need to run `eslint-plugin-testing-library` only for your testing files.

For doing this, you have set it up correctly in the `overrides` section:
```json
"overrides": [
    {
      "files": ["**/specs/**/*.[jt]s?(x)"],
      "extends": ["plugin:testing-library/react"],
      "rules": {
        "testing-library/no-node-access": "off",
        "testing-library/no-container": "off"
      }
    }
  ],
```

But you haven't removed `eslint-plugin-testing-library` from the main config!
```json5
 "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:testing-library/react", // <------- here you are still enabling the plugin for the whole codebase
    "prettier"
  ],
```

So if you remove `plugin:testing-library/react"` from your main `extends` property, no rule from this plugin will be run outside the tests (and you don't need to switch rules off in the main `rules` section).

I'm sorry this wasn't obvious from our ESLint `overrides` section, we tried to reflect it in the first step:
![CleanShot 2022-02-01 at 18 48 34@2x](https://user-images.githubusercontent.com/2677072/152023102-3a62b9e8-df09-485b-9db1-ce75ad79914e.png)

**Comment by TrueWill** [comment]
@Belco90 Perfect. Thank you!
